python/plugins/processing/algs/qgis/ExtentFromLayer.py

```diff
@@ -74,6 +77,11 @@ def __init__(self):
 
     def initAlgorithm(self, config=None):
         self.addParameter(QgsProcessingParameterMapLayer(self.INPUT, self.tr('Input layer')))
+        self.addParameter(QgsProcessingParameterNumber(self.ROUND_TO,
```

Comment:
This should be a distance parameter, linked to the INPUT parameter (since the units will match the layers unit). I think it should also be flagged as an advanced parameter.

Reply:
Thanks Nyall, I changed that and it looks better now with the units. I also added the advanced flag.

I'm not sure what the optional=True argument exactly does and if I should add it. There is a default value now, which in a pythonic way turns an argument in an optional one. But what does it change in processing. Just adding "[optional]" in the dialog? Or will it prevent errors for users who are already using this algorithm in their models? I would not want them to suffer from this change.

Reply:
The default value will always be used whenever a particular parameter value isn't specified. So there's no risk of breaking existing scripts or models by adding a new non-optional parameter, just so long as the algorithm can continue to function ok when using the default value of that new parameter.

In this case optional would mean "null values are ok!", which isn't the case.

Reply:
Ok, thanks for explaining. That should be OK then